### PR TITLE
Support purging of Managed HSMs

### LINKED ISSUE #2348: Managed HSMs are not purged
When running `azd down --force --purge`, Managed HSMs - a more secure, lifetime-billed Key Vault-like secure container - are not purged. These cost money even in deleted states and should be purged. Key Vaults, comparatively, are billed on a per-call basis so cost nothing when in a deleted state; though, if someone wants to reuse the same names, it can be a barrier.

### PR DESCRIPTION
**Pull request by heaths**
Fixes #2348

### CODE REVIEW

**Comment by heaths** [comment]
Also tested against live Key Vault and Managed HSM resources against https://github.com/heaths/azcrypto/pull/36.

I also fixed the casing - since "Key Vault" is the proper branding - in a couple UX-facing locations.

**Review by vhvb1989 — APPROVED** [review]
LGTM.
Just adding small question/suggestion

File: cli/azd/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### Bugs Fixed
 
+- [[2348]](https://github.com/Azure/azure-dev/pull/2279) Support purging Managed HSMs.
```
[the remark this answers is not in the document]
Comment (heaths) [reply cli/azd/CHANGELOG.md:11]: That's not what it is, though. Managed HSM is a different product that just happens to have the same keys API - by design - as Key Vault. Key Vaults with backing HSMs (premium SKU) is a separate feature of the Key Vault product.

**Comment by vhvb1989** [comment]
ping @ellismg .

**Review by ellismg — APPROVED** [review]
Thanks, @heaths!

**Pull request merged**